### PR TITLE
feat(sdk): track async tasks using WandbOperation

### PR DESCRIPTION
**Pull request by timoffex**
Description
---
Uses `wboperation` to track:

* Artifact operations ("uploading artifact {name}") with subtasks for each file
* FileStream operations ("uploading {history, summary, console logs}")
* `run.save()` operations ("uploading {file name}")

Since we don't print anything yet, Handler is updated to log operation stats at INFO level when receiving PollExit requests, at most once per minute.

### CODE REVIEW

File: core/pkg/server/sender_test.go [hunk 1]
```diff
@@ -46,7 +46,13 @@ func makeSender(client graphql.Client, recordChan chan *spb.Record, resultChan c
 	})
 	backend := server.NewBackend(logger, settings)
 	fileStream := server.NewFileStream(
-		backend, logger, observability.NewPrinter(), settings, nil)
+		backend,
+		logger,
+		nil,
+		observability.NewPrinter(),
+		settings,
+		nil,
```
Comment (kptkin) [inline core/pkg/server/sender_test.go:54]: add comments about the argument for the nil values

File: core/pkg/server/sender_test.go [hunk 2]
```diff
@@ -55,6 +61,7 @@ func makeSender(client graphql.Client, recordChan chan *spb.Record, resultChan c
 	runfilesUploader := server.NewRunfilesUploader(
 		runWork,
 		logger,
+		nil,
```
Comment (kptkin) [inline core/pkg/server/sender_test.go:64]: same as above

File: core/pkg/server/sender_test.go (hunk 1 above)
Comment (timoffex) [reply core/pkg/server/sender_test.go:54]: good point, done!

Comment (timoffex) [reply core/pkg/server/sender_test.go:54]: oops, just saw the commit for this in my local branch... forgot to squash it into the PR

#8582

**Pull request merged**